### DISCUSSION TITLE
AdGuardHome "Primary DNS" setting

### DISCUSSION THREAD

**Discussion by kevindd992002** [mimugmail/opn-repo#220] [Q&A]
Can you confirm what exactly does the "Primary DNS" setting in the AdGuardHome plugin do when AGH listens at port 53? I don't see any documentation about it and the single line help line there is not that helpful.

**Comment by mimugmail** [comment]
It needs this checkbox to register couple of settings of opnsense itself to use it which usually points to unbound

**Comment by kevindd992002** [comment]
Right but which one specifically? I'm curious which settings it specifically modifies when it points to unbound.

**Comment by mimugmail** [comment]
It just registers this variable like here with bind plugin, nothing more

https://github.com/opnsense/plugins/blob/master/dns/bind/src/etc/inc/plugins.inc.d/bind.inc#L67

**Reply by kevindd992002** [reply]
> It just registers this variable like here with bind plugin, nothing more
> 
> https://github.com/opnsense/plugins/blob/master/dns/bind/src/etc/inc/plugins.inc.d/bind.inc#L67

Sorry, I don't understand the code (since I mostly use Powershell). When it registers that variable, does it simply tell opnsense that AGH is now the local DNS resolver (instead of unbound) and let all system resolving work? What if you have it disabled? What will not work?

**Comment by kevindd992002** [comment]
@mimugmail do you have any updates on my question?

**Comment by mimugmail** [comment]
No, I'm not a core dev. I just added the checkbox cause it broke Setups when using AGH on port 53

**Reply by kevindd992002** [reply]
> No, I'm not a core dev. I just added the checkbox cause it broke Setups when using AGH on port 53

If that's the case, doesn't it make sense to remove this checkbox and just register the variable automatically when using AGH on port 53? What use case is there to have this disabled while using port 53?

**Reply by mimugmail** [reply]
You can run unbound as main resolver and AGH on a separate port and forward to there

**Reply by kevindd992002** [reply]
> You can run unbound as main resolver and AGH on a separate port and forward to there

Yes, of course. My point is that this setting should only be used when the listening port of AGH is port 53, otherwise it breaks. So why not set the registration "behind the scenes" instead of having it as an option.

There is no use case where you have that option disabled and AGH listening at port 53. Registering the variable automatically will prevent issues too.
